### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by Gh-Reza**
## Changes:

- Removed current_user function from ApplicationController
- Setup the devise
- Customized the login page and added the functionlity
- Added the email confirmation. Used litter_opener
- Added password reset functionality
- Modified the view

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <%= f.text_field :name, autocomplete: "name", autofocus: true, placeholder: "Name", class: "input" %>
+
+    </div>
+
+    <div class="field">
+      <%= f.email_field :email, autocomplete: "email", placeholder: "Email", class: "input"  %>
+    </div>
+
+    <div class="field">
+      <% if @minimum_password_length %>
```
Comment (V-Blaze) [inline app/views/devise/registrations/new.html.erb:16]: - You have done great so far :+1:  :100: But there are a few requirements you need to meet before moving to the next milestone. :+1: :smile: 
- [ ] When I start your app and go to `http://127.0.0.1:3000/users/sign_in`, I get the error message as shown in the screenshot below, I suggest you look at this and fix this as currently a user cant `sign_up or sing_in` to your app. :+1: :smile: :wave: 

- - Screenshot reference:
Screenshot from 2023-11-10 09-54-13

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Gh-Reza, @ahmed025566  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Installed and setup Devise gem✔
-  Correct git flow👋 
-  Readme looks professional ✔

 ## Required Changes ♻️
- [ ] @Gh-Reza, in your subsequent projects, when writing commit messages, Please use present imperative tense in commit messages. For example, `Add the default value for posts_counter` instead of `Added the default value for posts_counter`' Thanks! :wave: :small_airplane: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (Gh-Reza) [reply app/views/devise/registrations/new.html.erb:16]: Hello @V-Blaze, thank you for your time. I don't know what is the problem but it is working on my computer. I cloned the project into a different directory and it is working just fine.
image

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @ahmed025566  and @Gh-Reza     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**